platform_db_backup.sh

```diff
@@ -1,72 +1,69 @@
-    DB_NAME=$(echo $PLATFORM_RELATIONSHIPS | base64 --decode | jq -r ".${database}[0].path")
-    DB_USER=$(echo $PLATFORM_RELATIONSHIPS | base64 --decode | jq -r ".${database}[0].username")
-    DB_PASS=$(echo $PLATFORM_RELATIONSHIPS | base64 --decode | jq -r ".${database}[0].password")
-    DUMP_FOLDER=$HOME/drush-backups/${SITE}/$(date +%Y%m%d%H%M%S)
-    DUMP_FILE=${DUMP_FOLDER}/${DB_NAME}_$(date +%Y%m%d_%H%M%S).sql.gz
-    mkdir -p "${DUMP_FOLDER}"
-    echo "Creating database dump to ${DUMP_FILE}";
-    time mysqldump --max-allowed-packet=16M --single-transaction --skip-opt -e --quick --skip-disable-keys --skip-add-locks -a --add-drop-table --triggers --routines -h "${DB_HOST}" -P "${DB_PORT}" -u "${DB_USER}" -p"${DB_PASS}" "${DB_NAME}" | gzip -9 > "${DUMP_FILE}"
-    echo "DONE"
+  # special scenario created by having modular mount for files.
+  if [[ $SITE == '[^' ]]; then
+    continue;
+  fi
 
-  done
-
-  # loop through backups and upload to S3 until none left.
-  find $HOME/drush-backups -type f -name *gz -print | while read dump;
-  do
-    upload_dump_to_s3 $dump;
-  done
+  if [[ -v AWS_BACKUP_BUCKET ]]; then
+    if [ -d "files" ]; then
+      DAY=$(date -d "-1 day" +%Y-%m) && aws s3 sync ~/files/${SITE}/files s3://${AWS_BACKUP_BUCKET}/${PLATFORM_APPLICATION_NAME}/files-${SITE}/${PLATFORM_BRANCH}/files/${DAY}/ --storage STANDARD_IA
+    elif [ -d "web" ]; then
+      DAY=$(date -d "-1 day" +%Y-%m) && aws s3 sync ~/web/sites/${SITE}/files s3://${AWS_BACKUP_BUCKET}/${PLATFORM_APPLICATION_NAME}/files-${SITE}/${PLATFORM_BRANCH}/files/${DAY}/ --storage STANDARD_IA
+    else
+      DAY=$(date -d "-1 day" +%Y-%m) && aws s3 sync ~/docroot/sites/${SITE}/files s3://${AWS_BACKUP_BUCKET}/${PLATFORM_APPLICATION_NAME}/files-${SITE}/${PLATFORM_BRANCH}/files/${DAY}/ --storage STANDARD_IA
+    fi
+  elif [[ -v SFTP_SERVER ]]; then
+    DAY=$(date -d "-1 day" +%Y-%m)
+    # first create directory with files for current day in order to be able to move the folder to SFTP.
+    mkdir -p drush-backups/${PLATFORM_APPLICATION_NAME}/files-${SITE}/${PLATFORM_BRANCH}/files/${DAY}
+    echo "Uploading to SFTP server."
+    if [ -d "files" ]; then
+      tar -zcvf drush-backups/${PLATFORM_APPLICATION_NAME}/files-${SITE}/${PLATFORM_BRANCH}/files/${DAY}/files-${DAY}.tar.gz  files/${SITE}
+    elif [ -d "web" ]; then
+      tar -zcvf drush-backups/${PLATFORM_APPLICATION_NAME}/files-${SITE}/${PLATFORM_BRANCH}/files/${DAY}/files-${DAY}.tar.gz  web/sites/${SITE}/files
+    else
+      tar -zcvf drush-backups/${PLATFORM_APPLICATION_NAME}/files-${SITE}/${PLATFORM_BRANCH}/files/${DAY}/files-${DAY}.tar.gz  docroot/sites/${SITE}/files
+    fi
+  fi
+done
 
-  # clean up remaining files
-  find $HOME/drush-backups -mindepth 1 -type d -empty -delete
-fi
+# copy files from newly created directory to SFTP.
+scp -i .ssh/id_rsa.pub -P $SFTP_PORT -r ./drush-backups/${PLATFORM_APPLICATION_NAME} ${SFTP_USERNAME}@${SFTP_SERVER}:~/${SFTP_DIRECTORY}
```

Comment:
It looks like this will run regardless of the chosen backup server (it will run always). What if only aws is wanted?